app/src/androidTest/java/com/android/sample/screen/MainPageTest.kt

```diff
@@ -0,0 +1,85 @@
+package com.android.sample.screen
+
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performTouchInput
+import androidx.compose.ui.test.swipeLeft
+import androidx.compose.ui.unit.dp
+import androidx.test.espresso.intent.Intents
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Route
+import com.android.sample.ui.swipePage.ReceipeDisplay
+import com.android.sample.ui.swipePage.SwipePage
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+
+@RunWith(AndroidJUnit4::class)
+class MainPageTest : TestCase() {
+  private lateinit var navigationActions: NavigationActions
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    navigationActions = mock(NavigationActions::class.java)
+    `when`(navigationActions.currentRoute()).thenReturn(Route.MAIN)
+  }
+
+  @After
+  fun tearDown() {
+    Intents.release()
+  }
+
+  @Test
+  fun swipePageCorrectlyDisplayed() {
+    composeTestRule.setContent {
+      SwipePage(navigationActions) // Set up the SignInScreen directly
+    }
+    composeTestRule.onNodeWithTag("bottomNavigationMenu").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("topBar").assertIsDisplayed()
+  }
+
+  @Test
+  fun recipeAndDescriptionAreCorrectlyDisplayed() {
+    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }
+    composeTestRule.onNodeWithTag("recipeDescription").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("recipeImage").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("swipeUIDescription").assertIsDisplayed()
+  }
+
+  @Test
+  fun testBiggerDescription() {
+    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }
+
+    // Make sure the screen is displayed
+    composeTestRule.onNodeWithTag("displayDescription").assertIsDisplayed()
+
+    // Simulate a drag event
+    composeTestRule.onNodeWithTag("displayDescription").performClick()
+    composeTestRule.onNodeWithTag("RecipeEntireDescription").assertIsDisplayed()
+  }
+
+  @Test
+  fun testDraggingEvent() {
```

Comment:
### Review 1: Lack of Tests for Drag Events

**Review Comment:**
Currently, the tests only cover left swipe interactions. It would be beneficial to add tests for both left and right swipes to ensure full coverage of the swipe functionality.

### Review 2: Swipe Test Duration (5000 ms)

**Review Comment:**
The swipe duration in the `testDraggingEvent` test is set to 5000 ms, which might unnecessarily slow down test execution. Could you clarify why this specific duration is needed? It might be better to reduce the duration for faster execution, as long as the tests still simulate the swipe accurately.


**Code Snippet:**
```kotlin
@Test
fun testDraggingEventLeft() {
    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }

    // Make sure the screen is displayed
    composeTestRule.onNodeWithTag("draggableItem").assertIsDisplayed()

    // Simulate a left swipe event
    composeTestRule.onNodeWithTag("draggableItem").performTouchInput {
        swipeLeft(durationMillis = 1000) // Reduced duration for faster execution
    }
}

@Test
fun testDraggingEventRight() {
    composeTestRule.setContent { ReceipeDisplay(PaddingValues(0.dp)) }

    // Make sure the screen is displayed
    composeTestRule.onNodeWithTag("draggableItem").assertIsDisplayed()

    // Simulate a right swipe event
    composeTestRule.onNodeWithTag("draggableItem").performTouchInput {
        swipeRight(durationMillis = 1000) 
    }
}
```

Reply:
Thanks for pointing this out! I'll add a new test for right swiping and update the left swipe test accordingly